base_layer/core/src/validation/new/aggregate_body/chain_aggregate_body_validator.rs

```diff
@@ -0,0 +1,327 @@
+// Copyright 2022. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use std::convert::TryInto;
+
+use log::{trace, warn};
+use tari_common_types::types::{CommitmentFactory, HashOutput, PublicKey};
+use tari_crypto::keys::PublicKey as PublicKeyTrait;
+use tari_script::{ScriptContext, TariScript};
+use tari_utilities::hex::Hex;
+
+use crate::{
+    borsh::SerializedSize,
+    chain_storage::{BlockchainBackend, BlockchainDatabase, MmrTree},
+    consensus::ConsensusConstants,
+    transactions::{
+        transaction_components::{Transaction, TransactionError, TransactionInput, TransactionOutput},
+        CryptoFactories,
+    },
+    validation::ValidationError,
+};
+
+pub const LOG_TARGET: &str = "c::val::chain_aggregate_body_validator";
+
+pub struct ChainLinkedAggregateBodyValidator<B> {
+    db: BlockchainDatabase<B>,
+    factories: CryptoFactories,
+}
+
+impl<B: BlockchainBackend + 'static> ChainLinkedAggregateBodyValidator<B> {
+    pub fn new(db: BlockchainDatabase<B>) -> Self {
+        Self {
+            db,
+            factories: CryptoFactories::default(),
+        }
+    }
+
+    pub async fn validate(
+        &self,
+        tx: &Transaction,
```

Comment:
```suggestion
        body: &AggregateBody,
```
make it body
These tests will need to be done by the mempool and block validation. 
Block validation does not have the tx anymore.

Reply:
We need all the helper functions to be the same code/function